### ISSUE TITLE
`unused_closure_parameter`: fails to detect escaped identifiers.

### ISSUE THREAD

**Issue by lhunath** [realm/SwiftLint#3706]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Swiftlint's unused closure parameter rule fails to detect uses of the identifier in contexts where it is quoted with single-ticks. This is particularly crucial for identifiers that have the same name as keywords. The result is false-positives.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ cat test.swift
let _: (String) -> String = { `break` in `break` }
$ swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
test.swift:1:31: warning: Unused Closure Parameter Violation: Unused parameter "break" in a closure should be replaced with _. (unused_closure_parameter)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 12.5.1, Build version 12E507

**Comment by lhunath** [comment]
Didn't do a good job searching.
https://github.com/realm/SwiftLint/issues/3628
